transcom/mymove
# B 18475 service item status change on shipment address update approval

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A888121&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

This PR changes the status of destination SIT service items when the TOO approves a shipment address update request. Since the update of a shipment's `destination_address` changes the destination SIT service item's `final_destination_address` to be the same - it is expected that if the shipment's address change requires TOO approval, then the service item will meet specific checks.

We want the status of the service item to change to trigger TOO review. With the completion of this story, a ToolTip component will show displaying the changed values of the service items.

### How to test

1. Access MM as the Prime and find a shipment containing destination SIT service items
2. Request to update a shipment's destination address (make sure you update the address to trigger TOO review - this is easiest by changing it to be something far away from what it previously was)
3. Review the request as a TOO
4. Approve the request
5. Navigate to the MTO page
6. The service items should be in the REQUESTED section since their status has changed

Head on over to the prime sim and update the shipment destination addy
![Screenshot 2024-01-16 at 12 56 06 PM](https://github.com/transcom/mymove/assets/136510600/c0c8c66c-840a-4775-a22a-f1b8a9d6cc6d)

Fill out the form and send it
![Screenshot 2024-01-16 at 12 57 06 PM](https://github.com/transcom/mymove/assets/136510600/0738ce2d-d638-4917-b080-dff1e3092d2f)

Log in as the TOO and go edit the shipment (should see the flag)
![Screenshot 2024-01-16 at 12 58 03 PM](https://github.com/transcom/mymove/assets/136510600/f58fb6c0-5129-497f-9d21-823c0a29b411)

Review the request
![Screenshot 2024-01-16 at 12 58 20 PM](https://github.com/transcom/mymove/assets/136510600/47a2db27-8228-41c9-941c-bca8bbe9229b)

Fill out the form and approve it
![Screenshot 2024-01-16 at 12 58 51 PM](https://github.com/transcom/mymove/assets/136510600/09cee054-43b5-45fb-affb-70c06f15bd25)

Go over to the MTO page
![Screenshot 2024-01-16 at 12 59 19 PM](https://github.com/transcom/mymove/assets/136510600/4234bee8-096e-49d0-a990-b303ea89181d)

You should now be required to approve/reject the destination SIT service items
![Screenshot 2024-01-16 at 12 59 44 PM](https://github.com/transcom/mymove/assets/136510600/053ae4d4-57bc-4437-af72-89d34252e731)
